kafka-client/src/test/java/dev/responsive/kafka/integration/ResponsiveSessionStoreIntegrationTest.java

```diff
@@ -0,0 +1,300 @@
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.common.serialization.LongDeserializer;
+import org.apache.kafka.common.serialization.LongSerializer;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.Materialized;
+import org.apache.kafka.streams.kstream.SessionWindows;
+import org.apache.kafka.streams.kstream.Windowed;
+import org.apache.kafka.streams.kstream.internals.SessionWindow;
+import org.apache.kafka.streams.state.SessionStore;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class ResponsiveSessionStoreIntegrationTest {
+
+  @RegisterExtension
+  static ResponsiveExtension EXTENSION = new ResponsiveExtension(StorageBackend.MONGO_DB);
+
+  private static final String INPUT_TOPIC = "input";
+  private static final String OTHER_TOPIC = "other";
+  private static final String OUTPUT_TOPIC = "output";
+
+  private final Map<String, Object> responsiveProps = new HashMap<>();
+
+  private String name;
+  private Admin admin;
+
+  @BeforeEach
+  public void before(
+      final TestInfo info,
+      final Admin admin,
+      @ResponsiveConfigParam final Map<String, Object> responsiveProps
+  ) throws InterruptedException, ExecutionException {
+    // append a random int to avoid naming collisions on repeat runs
+    name = info.getTestMethod().orElseThrow().getName() + "-" + new Random().nextInt();
+
+    this.responsiveProps.putAll(responsiveProps);
+
+    this.admin = admin;
+    final var result = admin.createTopics(
+        List.of(
+            new NewTopic(inputTopic(), Optional.of(1), Optional.empty()),
+            new NewTopic(otherTopic(), Optional.of(1), Optional.empty()),
+            new NewTopic(outputTopic(), Optional.of(1), Optional.empty())
+        )
+    );
+    result.all().get();
+  }
+
+  @Test
+  public void shouldComputeSessionAggregate() throws Exception {
+    // Given:
+    final Duration inactivityGap = Duration.ofSeconds(5);
+    final Duration gracePeriod = Duration.ofSeconds(2);
+    final SessionWindows window =
+        SessionWindows.ofInactivityGapAndGrace(inactivityGap, gracePeriod);
+
+    final Materialized<String, String, SessionStore<Bytes, byte[]>> store =
+        ResponsiveStores.sessionMaterialized(
+            ResponsiveSessionParams.session(name, inactivityGap, gracePeriod)
+        );
+
+    // Start from timestamp of 0L to get predictable results
+    final List<KeyValueTimestamp<String, String>> inputEvents = asList(
```

Comment:
can we test things outside the inactivity gap and also inside the inactivity gap but requiring the grace period (i.e. out of order events)?